src/Plugins/ResizeMirror/ResizeMirror.js

```diff
@@ -0,0 +1,109 @@
+import AbstractPlugin from 'shared/AbstractPlugin';
+
+const onDragOver = Symbol('onDragOver');
+
+/**
+ * ResizeMirror default options
+ * @property {Object} defaultOptions
+ * @property {Number} defaultOptions.duration
+ * @property {String} defaultOptions.easingFunction
+ * @type {Object}
+ */
+export const defaultOptions = {
+  duration: 150,
+  easingFunction: 'ease-in-out',
+};
+
+/**
+ * ResizeMirror plugin resizes the mirror element based on the draggable elements it's hovering over
+ * @class ResizeMirror
+ * @module ResizeMirror
+ * @extends AbstractPlugin
+ */
+export default class ResizeMirror extends AbstractPlugin {
+
+  /**
+   * ResizeMirror constructor.
+   * @constructs ResizeMirror
+   * @param {Draggable} draggable - Draggable instance
+   */
+  constructor(draggable) {
+    super(draggable);
+
+    /**
+     * ResizeMirror options
+     * @property {Object} options
+     * @property {Number} defaultOptions.duration
+     * @property {String} defaultOptions.easingFunction
+     * @type {Object}
+     */
+    this.options = {
+      ...defaultOptions,
+      ...this.getOptions(),
+    };
+
+    this[onDragOver] = this[onDragOver].bind(this);
+  }
+
+  /**
+   * Attaches plugins event listeners
+   */
+  attach() {
+    this.draggable
+      .on('drag:over', this[onDragOver])
+      .on('drag:over:container', this[onDragOver]);
+  }
+
+  /**
+   * Detaches plugins event listeners
+   */
+  detach() {
+    this.draggable
+      .off('drag:over', this[onDragOver])
+      .off('drag:over:container', this[onDragOver]);
+  }
+
+  /**
+   * Returns options passed through draggable
+   * @return {Object}
+   */
+  getOptions() {
+    return this.draggable.options.resizeMirror || {};
+  }
+
+  /**
+   * Drag over handler
+   * @param {DragOverEvent} dragEvent
+   * @private
+   */
+  [onDragOver](dragEvent) {
+    requestAnimationFrame(animateResize({
+      dragEvent,
+      draggableSelector: this.draggable.options.draggable,
+      originalSourceClassName: this.draggable.getClassNameFor('source:original'),
+      options: this.options,
+    }));
+  }
+}
+
+function animateResize({dragEvent, draggableSelector, originalSourceClassName, options}) {
+  return () => {
+    const overElement = dragEvent.over || dragEvent.overContainer.querySelector(`${draggableSelector}:not(.${originalSourceClassName})`);
+
+    if (overElement) {
+      dragEvent.overContainer.appendChild(dragEvent.mirror);
+      const overRect = overElement.getBoundingClientRect();
+
+      dragEvent.mirror.style.width = `${overRect.width}px`;
+      dragEvent.mirror.style.height = `${overRect.height}px`;
+      dragEvent.mirror.style.transition = `width ${options.duration}ms ${options.easingFunction},` +
```

Comment:
This isn't going to work - you are applying the `transition` logic at the same time as the dimension change. Simply put - its not going to transition. This needs to be added before... why not just add it on `start`.

That being said - this does not take into consideration any `transition` styles already applied to that element. If you want to take this approach, you would have to `getComputedStyle` and "add on" to the list of existing `transition` properties.

In general, I'm against this kind of animation hijacking. I personally think that its better to JUST change dimensions and not apply a transition. That should be up to the consumer. That way, I can customize the easing and timing to my liking using CSS.

What do you think?